haystack_experimental/components/agents/agent.py

```diff
@@ -0,0 +1,264 @@
+        system_prompt: Optional[str] = None,
+        api_key: Secret = Secret.from_env_var("LLM_API_KEY", strict=False),
+        exit_condition: str = "text",
+        generation_kwargs: Optional[Dict[str, Any]] = None,
+        state_schema: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initialize the agent component.
+
+        :param model: Model identifier in the format "provider:model_name"
+        :param generation_kwargs: Keyword arguments for the chat model generator
+        :param tools: List of Tool objects available to the agent
+        :param exit_condition: Either "text" if the agent should return when it generates a message without tool calls
+            or the name of a tool that will cause the agent to return once the tool was executed
+        :param input_variables: Dictionary mapping input variable names to their types
+        :param output_variables: Dictionary mapping output variable names to their types
+        :raises ValueError: If model string format is invalid or exit_condition is not valid
+        """
+        if ":" not in model:
+            raise ValueError("Model string must be in format 'provider:model_name'")
+
+        provider, _ = model.split(":")
+        if provider not in _PROVIDER_GENERATOR_MAPPING:
+            raise ValueError(
+                f"Provider must be one of {list(_PROVIDER_GENERATOR_MAPPING.keys())}"
+            )
+
+
+        valid_exits = ["text"] + [tool.name for tool in tools or []]
+        if exit_condition not in valid_exits:
+            raise ValueError(f"Exit condition must be one of {valid_exits}")
+
+        if state_schema is not None:
+            _validate_schema(state_schema)
+        self.state_schema = state_schema or {}
+
+        self.model = model
+        self.generation_kwargs = generation_kwargs or {}
+        self.tools = tools or []
+        self.system_prompt = system_prompt
+        self.exit_condition = exit_condition
+        self.api_key = api_key
+
+        component.set_input_type(instance=self, name="messages", type=List[ChatMessage])
+        output_types = {"messages": List[ChatMessage]}
+        for param, config in self.state_schema.items():
+            component.set_input_type(self, name=param, type=config["type"], default=None)
+            output_types[param] = config["type"]
+        component.set_output_types(self, **output_types)
+
+        self._initialize_pipeline()
+
+    def _initialize_pipeline(self) -> None:
+        """Initialize the component pipeline with all necessary components and connections."""
+        provider, model_name = self.model.split(":")
+
+        if provider == "anthropic":
+            anthropic_import.check()
+
+        # Initialize components
+        generator = _PROVIDER_GENERATOR_MAPPING[provider](
+            model=model_name,
+            tools=self.tools,
+            api_key=self.api_key,
+            generation_kwargs=self.generation_kwargs,
+        )
```

Comment:
I think it would be good to make sure we add a `warm_up` method to the `Agent` which calls all potentially relevant `warm_up` methods of it's subsequent components. For example, some ChatGenerators could have warm_up methods (e.g. only initialize the client at warm_up time). 

Maybe this is also relevant for tools? Not entirely sure how we handle tools that are Pipelines.